components/custom-checkbox/css/_mixin.scss

```diff
@@ -3,107 +3,123 @@
+  border-width: var(--utrecht-custom-checkbox-border-width);
   height: var(--utrecht-custom-checkbox-size);
-  justify-content: center;
-  left: 0;
-  pointer-events: none;
-  position: absolute;
-  top: 0;
+  margin-block-start: var(--utrecht-custom-checkbox-margin-block-start);
+  -webkit-print-color-adjust: exact;
+  print-color-adjust: exact;
+  vertical-align: top;
   width: var(--utrecht-custom-checkbox-size);
-  z-index: 1000;
 }
 
-@mixin utrecht-custom-checkbox__box--checked {
+@mixin utrecht-custom-checkbox--checked {
   background-color: var(
     --utrecht-custom-checkbox-checked-background-color,
     var(--utrecht-custom-checkbox-background-color)
   );
+  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 20'%3e%3cpath fill='none' stroke='%23fff' stroke-linecap='round' stroke-linejoin='round' stroke-width='3' d='m6 10 3 3 6-6'/%3e%3c/svg%3e");
   border-color: var(--utrecht-custom-checkbox-checked-border-color, var(--utrecht-custom-checkbox-border-color));
   border-width: var(--utrecht-custom-checkbox-checked-border-width, var(--utrecht-custom-checkbox-border-width));
-  color: var(--utrecht-custom-checkbox-checked-color, var(--utrecht-custom-checkbox-color));
 }
 
-@mixin utrecht-custom-checkbox__box--disabled {
-  background-color: var(
-    --utrecht-custom-checkbox-disabled-background-color,
-    var(--utrecht-custom-checkbox-background-color)
-  );
+@mixin utrecht-custom-checkbox--disabled {
+  &:checked {
+    background-color: var(
+      --utrecht-custom-checkbox-disabled-background-color,
+      var(--utrecht-custom-checkbox-background-color)
+    );
+    border-color: var(--utrecht-custom-checkbox-disabled-border-color, var(--utrecht-custom-checkbox-border-color));
+  }
+  &:active {
+    border-color: var(--utrecht-custom-checkbox-disabled-border-color, var(--utrecht-custom-checkbox-border-color));
+  }
+
   border-color: var(--utrecht-custom-checkbox-disabled-border-color, var(--utrecht-custom-checkbox-border-color));
   border-width: var(--utrecht-custom-checkbox-disabled-border-width, var(--utrecht-custom-checkbox-border-width));
-  color: var(--utrecht-custom-checkbox-disabled-color, var(--utrecht-custom-checkbox-color));
   cursor: var(--utrecht-action-disabled-cursor);
 }
 
-@mixin utrecht-custom-checkbox__box--indeterminate {
-  color: var(--utrecht-custom-checkbox-indeterminate-color, var(--utrecht-custom-checkbox-color));
-}
-
-@mixin utrecht-custom-checkbox__box--invalid {
+@mixin utrecht-custom-checkbox--indeterminate {
   background-color: var(
-    --utrecht-custom-checkbox-invalid-background-color,
+    --utrecht-custom-checkbox-indeterminate-background-color,
     var(--utrecht-custom-checkbox-background-color)
   );
+  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 20'%3e%3cpath fill='none' stroke='%23fff' stroke-linecap='round' stroke-linejoin='round' stroke-width='3' d='M6 10h8'/%3e%3c/svg%3e");
```

Comment:
All the mixins (and mixins in general) use variables in some way and don't contain hard-coded asset references or property values. Except for these data URLs. 

Is it possible to keep the SVG image in the markup instead?  
Using a single icon component for the multiple states. 
The other option is to move the data value to the proprietary tokens. That way, the icons remain themable.

Reply:
I think this approach with the hardcoded `<svg>` was meant to achieve the desired styling with only `<input type="checkbox">`. For the Web Component/React etc variant having the SVG in the markup is absolutely fine. For web components I guess it would be nice to able to customize the icon with `<svg slot="checkmark-checked"/>` or something like that.